src/components/common/animation-control/animation-control.js

```diff
@@ -0,0 +1,197 @@
+        this._animation = requestAnimationFrame(this._nextFrame);
+      }
+    }
+
+    onSlider1Change = val => {
+      const {domain} = this.props.animationConfig;
+      if (val >= domain[0] && val <= domain[1]) {
+        this.props.updateAnimationTime(val);
+      }
+    };
+
+    _updateAnimationTime = () => {
+      const {domain} = this.props.animationConfig;
+      this.props.updateAnimationTime(domain[0]);
+      this._startAnimation();
+    };
+
+    _startAnimation = () => {
+      this._pauseAnimation();
+      this.setState({isAnimating: true});
+    };
+
+    _nextFrame = () => {
+      this._animation = null;
+      const {currentTime, domain, speed} = this.props.animationConfig;
+      const adjustedSpeed = ((domain[1] - domain[0]) / BASE_SPEED) * speed;
+      const nextTime =
+        currentTime + speed > domain[1]
+          ? domain[0]
+          : currentTime + adjustedSpeed;
+      this.props.updateAnimationTime(nextTime);
+    };
+
+    _pauseAnimation = () => {
+      if (this._animation) {
+        cancelAnimationFrame(this._animation);
+        this._animation = null;
+      }
+      this.setState({isAnimating: false});
+    };
+
+    toggleSpeedControl = () => {
+      this.setState({showSpeedControl: !this.state.showSpeedControl});
+    };
+
+    onChange = () => {
+      this.toggleSpeedControl();
+    };
+
+    render() {
+      const {currentTime, domain, speed} = this.props.animationConfig;
+      const {showSpeedControl} = this.state;
+
+      return (
+        <BottomWidgetInner className="bottom-widget--inner">
+          <AnimationWidgetInner className="animation-widget--inner">
+            <div style={{marginLeft: '-10px'}}>
+              <AnimationPlaybacks
+                className="animation-control-playpause"
+                startAnimation={this._startAnimation}
+                isAnimating={this.state.isAnimating}
+                pauseAnimation={this._pauseAnimation}
+                resetAnimation={this._resetAnimation}
+                buttonHeight={BUTTON_HEIGHT}
+                buttonStyle="link"
+              />
+            </div>
+            <StyledDomain className="animation-control__time-domain">
+              <span>{moment.utc(domain[0]).format(defaultTimeFormat)}</span>
+            </StyledDomain>
+            <SliderWrapper className="animation-control__slider">
+              <Slider
+                showValues={false}
+                isRanged={false}
+                minValue={domain ? domain[0] : 0}
+                maxValue={domain ? domain[1] : 1}
+                value1={currentTime}
+                onSlider1Change={this.onSlider1Change}
+                enableBarDrag={true}
+              />
+            </SliderWrapper>
+            <StyledDomain className="animation-control__time-domain">
+              <span>{moment.utc(domain[1]).format(defaultTimeFormat)}</span>
+            </StyledDomain>
+            <div style={{marginRight: '-10px'}}>
```

Comment:
constant for the value of the margin. 
Another improvement would be to define the entire object as constant rather than inline